src/Collection/CollectionTrait.php

```diff
@@ -313,6 +319,12 @@ public function indexBy(callable|string $path): CollectionInterface
                     'Use a callback to return a default value for that path.'
                 );
             }
+            if ($pathValue instanceof BackedEnum) {
+                $pathValue = $pathValue->value;
+            } elseif ($pathValue instanceof UnitEnum) {
```

Comment:
We don't support `UnitEnum` in `combine()` and raise an error if one is encountered. It feels inconsistent if `UnitEnum` sometimes works and sometimes doesn't.

Reply:
I fixed that in the same PR, see description above. I agree that it should be consistent

Reply:
Oops, I missed that.